wdl-lint/src/rules/redundant_input_assignment.rs

```diff
@@ -0,0 +1,113 @@
+//! A lint rule for flagging redundant input assignments
+
+use std::fmt::Debug;
+
+use rowan::ast::AstNode;
+use wdl_ast::AstNodeExt;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::v1::CallStatement;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Redundant Input Assignment rule.
+const ID: &str = "RedundantInputAssignment";
+
+/// Create a "Redundant Input Assignment" diagnostic.
+fn redundant_input_assignment(span: Span, name: &str) -> Diagnostic {
+    Diagnostic::note("redundant input assignment can be shortened")
+        .with_rule(ID)
+        .with_label(
+            format!("redundant input assignment can be shortened to `{}`", name),
+            span,
+        )
+}
```

Comment:
```suggestion
fn redundant_input_assignment(span: Span, name: &str) -> Diagnostic {
    Diagnostic::note("redundant input assignment")
        .with_rule(ID)
        .with_highlight(
            span,
        ).with_fix(
            format!("can be shortened to `{}`", name),
        )
}
```
I think this is a place where it makes sense to use a `fix`, despite that open issue saying we should use those more sparingly (#193)

Reply:
I just typed this in GH it might not compile lol. I think what I was going for is clear though

Reply:
Missed the exact language at first, but just re-committed with this exact verbiage